scripts/src/java/org/oppia/android/scripts/docs/KDocValidityCheck.kt

```diff
@@ -122,17 +140,23 @@ private fun createKtFile(codeString: String, fileName: String): KtFile {
  * @param file the file to be checked
  * @param kDocNotRequiredAnnotationEntryList the list of annotation entries which when present
  *     on an element, the element does not needs to be checked for a KDoc.
- * @return whether the file has a KDoc presence failure
+ * @return Returns list of elements in the file which are missing KDoc
  */
-private fun hasKdocFailure(file: File, kDocNotRequiredAnnotationEntryList: List<String>): Boolean {
+private fun hasKdocFailure(
+  file: File,
+  kDocNotRequiredAnnotationEntryList: List<String>
+): List<Pair<File, Int?>?> {
   val ktFile = createKtFile(file.asString(), file.name)
-  return ktFile.children.fold(initial = false) { isFailing, elem ->
-    val childFailingKdoc = if (elem is KtElement) elementIterator(
-      elem,
-      file,
-      kDocNotRequiredAnnotationEntryList
-    ) else false
-    isFailing || childFailingKdoc
+  return ktFile.children.flatMap { elem ->
+    if (elem is KtElement) {
+      recursiveKdocPresenceChecker(
+        elem,
+        file,
+        kDocNotRequiredAnnotationEntryList
+      )
+    } else {
+      listOf(null)
```

Comment:
Why not return empty list, instead? Ditto below.

Prefer to avoid nullable unless needed.

Reply:
Done.